shaders/pbr-frag.glsl

```diff
@@ -53,257 +66,237 @@ varying vec3 v_Normal;
+    return attenuationL * attenuationV;
 }
 
 // The following equation(s) model the distribution of microfacet normals across the area being drawn (aka D())
-// implementation from “Average Irregularity Representation of a Roughened Surface for Ray Reflection” by T. S. Trowbridge, and K. P. Reitz
-float GGX(PBRInfo pbrInputs)
+// Implementation from "Average Irregularity Representation of a Roughened Surface for Ray Reflection" by T. S. Trowbridge, and K. P. Reitz
+// Follows the distribution function recommended in the SIGGRAPH 2013 course notes from EPIC Games [1], Equation 3.
+float microfacetDistribution(PBRInfo pbrInputs)
 {
-  float roughnessSq = pbrInputs.alphaRoughness*pbrInputs.alphaRoughness;
-  float f = (pbrInputs.NdotH * roughnessSq - pbrInputs.NdotH) * pbrInputs.NdotH + 1.0;
-  return roughnessSq / (M_PI * f * f);
+    float roughnessSq = pbrInputs.alphaRoughness * pbrInputs.alphaRoughness;
+    float f = (pbrInputs.NdotH * roughnessSq - pbrInputs.NdotH) * pbrInputs.NdotH + 1.0;
+    return roughnessSq / (M_PI * f * f);
 }
 
 void main()
 {
-  // Normal Map
-  #ifndef HAS_TANGENTS
-  vec3 pos_dx = dFdx(v_Position);
-  vec3 pos_dy = dFdy(v_Position);
-  vec3 tex_dx = dFdx(vec3(v_UV, 0.0));
-  vec3 tex_dy = dFdy(vec3(v_UV, 0.0));
-  vec3 t = (tex_dy.t * pos_dx - tex_dx.t * pos_dy) / (tex_dx.s * tex_dy.t - tex_dy.s * tex_dx.t);
-
-  #ifdef HAS_NORMALS
-  vec3 ng = normalize(v_Normal);
-  #else
-  vec3 ng = cross(pos_dx, pos_dy);
-  #endif
-
-  t = normalize(t - ng * dot(ng, t));
-  vec3 b = normalize(cross(ng, t));
-  mat3 tbn = mat3(t, b, ng);
-  #else // HAS_TANGENTS && HAS_NORMALS
-  mat3 tbn = v_TBN;
-  #endif
-
-  #ifdef HAS_NORMALMAP
-  vec3 n = texture2D(u_NormalSampler, v_UV).rgb;
-  n = normalize(tbn * ((2.0 * n - 1.0) * vec3(u_NormalScale, u_NormalScale, 1.0)));
-  #else
-  vec3 n = tbn[2].xyz;
-  #endif
-
-  vec3 v = normalize(u_Camera - v_Position);
-  vec3 l = normalize(u_LightDirection);
-  vec3 h = normalize(l+v);
-  vec3 reflection = -normalize(reflect(v, n));
-
-  float NdotL = clamp(dot(n, l), 0.001, 1.0);
-  float NdotV = abs(dot(n, v)) + 0.001;
-  float NdotH = clamp(dot(n, h), 0.0, 1.0);
-  float LdotH = clamp(dot(l, h), 0.0, 1.0);
-  float VdotH = clamp(dot(v, h), 0.0, 1.0);
-
-  float perceptualRoughness = u_MetallicRoughnessValues.y;
-  float metallic = u_MetallicRoughnessValues.x;
-  #ifdef HAS_METALROUGHNESSMAP
-  vec4 mrSample = texture2D(u_MetallicRoughnessSampler, v_UV);
-  perceptualRoughness = mrSample.g * perceptualRoughness;
-  metallic = mrSample.b * metallic;
-  #endif
-
-  perceptualRoughness = clamp(perceptualRoughness, c_MinRoughness, 1.0);
-  metallic = clamp(metallic, 0.0, 1.0);
-
-  #ifdef HAS_BASECOLORMAP
-  vec4 baseColor = texture2D(u_BaseColorSampler, v_UV) * u_BaseColorFactor;
-  #else
-  vec4 baseColor = u_BaseColorFactor;
-  #endif
-
-  vec3 f0 = vec3(0.04);
-  // is this the same? test!
-  vec3 diffuseColor = mix(baseColor.rgb * (1.0 - f0), vec3(0.0, 0.0, 0.0), metallic);
```

Comment:
Hmmm.  I took another look at comparing master against this branch, and I'm not seeing a difference.  There must have been some incremental change where something was broken and then serendipitously fixed at the time.  Sorry for the distraction and thanks for verifying.